feat(serverless): add comparaison table (#4031)
scaleway/docs-content: macros/serverless/difference-jobs-functions-containers.mdx

## Differences between Functions, Containers, and Jobs

### Considerations

* **Serverless Containers** offers more flexibility by allowing you to deploy any containerized application. It's suitable for stateless services and can handle more complex applications with configurable concurrency and no maximum execution time.

> * **Serverless Functions** is ideal for lightweight, event-driven code where you don't manage the underlying infrastructure. It's limited to specific languages and is not suitable for long-running processes.

Reviewer: ```suggestion
* **Serverless Functions** is ideal for lightweight, event-driven code where you do not manage the underlying infrastructure. It is limited to specific languages and not suitable for long-running processes.
```

> * **Serverless Jobs** is designed for executing batch jobs or tasks that need to run to completion. It is ideal for data processing tasks, migrations, or any workload that doesn't fit the request-response model.

Reviewer: ```suggestion
* **Serverless Jobs** is designed for executing batch jobs or tasks that need to run to completion. It is ideal for data processing tasks, migrations, or any workload that does not fit the request-response model.
```

> ### Comparaison table

Reviewer: ```suggestion
### Comparison table
```

| **Criteria**                       | **Serverless Functions**                                            | **Serverless Containers**                                           | **Serverless Jobs**                                     |
|------------------------------------|---------------------------------------------------------------------|---------------------------------------------------------------------|---------------------------------------------------------|
| **Use case**                       | Event-driven functions responding to events or HTTP requests        | Stateless HTTP containers handling web services or APIs             | Processing tasks that run to completion                 |
| **Scaling**                        | Automatically scales based on incoming events                       | Automatically scales based on incoming events                       | Manually specified number of job executions             |

> | **Maximun execution time**         | Up to 15 minutes per request                                        | Up to 15 minutes per request                                        | 24 hours                                                |

Reviewer 2: ```suggestion
| **Maximum execution time**         | Up to 15 minutes per request                                        | Up to 15 minutes per request                                        | 24 hours                                                |
```

| **Concurrency**                    | Single request per instance                                         | Configurable concurrency per instance                               | Not applicable                                          |

> | **Pricing model**                  | Pay per exeuction time and invocations                              | Pay per vCPU and memory usage                                       | Pay per vCPU and memory usage                           |

Reviewer 2: ```suggestion
| **Pricing model**                  | Pay based on exeuction time and invocations                              | Pay based on vCPU and memory usage                                       | Pay based on vCPU and memory usage                           |
```

| **Triggers**                       | HTTP requests, scheduled (CRONs), messaging events (NATS, SQS, SNS) | HTTP requests, scheduled (CRONs), messaging events (NATS, SQS, SNS) | Manual invocation or scheduled                          |
| **Language support**               | Node, PHP, Python, Go, Rust                                         | Any language/runtime within a container                             | Any language/runtime within a container                 |

> | **Infrastructure management**      | Fully managed, handles provisionning and scaling automatically      | Fully managed, handles provisionning and scaling automatically      | Fully managed, abstracts away infrastructure management |

Reviewer 2: ```suggestion
| **Infrastructure management**      | Fully managed, handles provisioning and scaling automatically      | Fully managed, handles provisionnng and scaling automatically      | Fully managed, abstracts away infrastructure management |
```

> | **Cold-starts**                    | Can have startup latency on new instances                           | Can have startup latency on new instances                           | No applicable (jobs starts when scheduled or invoked)   |

Reviewer 2: ```suggestion
| **Cold-starts**                    | There may be startup latency on new instances                           | There may be startup latency on new instances                           | Not applicable (jobs starts when scheduled or invoked)   |
```

| **Supports custom containers**     | No, deploy code in supported languages only                         | Yes, deploy any containerized application                           | Yes, deploy any containerized application               |

> | **Long-running Processes**         | Not suitable for long running tasks                                 | Not suitable for long running tasks                                 | Ideal for long running processes                        |

Reviewer 2: ```suggestion
| **Long-running processes**         | Not suitable for long running tasks                                 | Not suitable for long running tasks                                 | Ideal for long running processes                        |
```
